docs/docs/design/statement-distribution.md

```diff
@@ -0,0 +1,134 @@
+# Statement Distribution
+
+This subsystem is responsible for distribute signed statements that we have generated and forwarding statements generated by our peers.
+
+## Subsystem Structure
+
+The subsystem needs to handle couple channels, beyond the overseer channel we also need to manage request and responses channels that are generated by spawned goroutines, these goroutines are called "responders" and in the parity's implementation there exists 2 "responders", *larger_statement_responder* (legacy) and *candidate_responder* (v2).
+
+The responsability of these "responders" are to listen for incoming requests, foward them to inside the statement distribution (as they are spawned goroutines), which will reach the target function that will respond the request properly, the "responders" also act like a limiter on the amount of requests we are capable to handle.
+ 
+For V2:
+- Incoming request type [AttestedCandidateResponse](https://github.com/paritytech/polkadot-sdk/blob/3d8da815ecd12b8f04daf87d6ffba5ec4a181806/polkadot/node/network/protocol/src/request_response/v2.rs#L32)
+- Response type [AttestedCandidateResponse](https://github.com/paritytech/polkadot-sdk/blob/3d8da815ecd12b8f04daf87d6ffba5ec4a181806/polkadot/node/network/protocol/src/request_response/v2.rs#L47)
+- The handler [answer_request](https://github.com/paritytech/polkadot-sdk/blob/c0921339f9d486981b3681760ee83ba9237f2eaa/polkadot/node/network/statement-distribution/src/v2/mod.rs#L3245)
+
+For V1 (No need for handling such requests or implementing the handlers, will be removed soon see https://github.com/paritytech/polkadot-sdk/issues/4447):
+- Incoming request type [StatementFetchingRequest](https://github.com/paritytech/polkadot-sdk/blob/c0921339f9d486981b3681760ee83ba9237f2eaa/polkadot/node/network/protocol/src/request_response/v1.rs#L196)
+- The handler [handle_responder_message](https://github.com/paritytech/polkadot-sdk/blob/c0921339f9d486981b3681760ee83ba9237f2eaa/polkadot/node/network/statement-distribution/src/legacy_v1/mod.rs#L1896)
+
+The V2 brings a new advantage, the validators don't send each other heavy `CommitedCandidateReceipt`, but instead request them lazily through request/response protocols.
+
+The messages that validators can exchange are: `Statement`, contains only a signed compact statement, `BackedCandidateManifest`, advertise a description of a backed candidate and stored statements and `BackedCandidateAcknowledgement`, acknowledges that a backed candidate is fully known.
+
+### Messages Received
+
+The subsystem must be registered with the overseer and handle two subsystem-specific messages from it:
```

Comment:
In this list, sometimes the description is above and sometimes it is below, can we be consistent with the placement